### PR TITLE
Add cache config to service base

### PR DESCRIPTION
**Pull request by KrishnaIyer**
#### Summary

Add a Cache to the `ServiceBase`. This enables the configuration of a separate redis for the IS cache.

#### Changes

- Add a cache (currently Redis) to `ServiceBase`.
- Use this in the IS Cache setting.

#### Notes for Reviewers

- Either this file has not been gofumpted or my tooling has issues.
- This kinda breaks config right?

### CODE REVIEW

File: pkg/config/shared.go [hunk 1]
```diff
@@ -139,6 +139,12 @@ type CloudEvents struct {
 	SubscribeURL string `name:"subscribe-url" description:"URL for the subscription to receiving events"`
 }
 
+// Cache represents configuration for a caching system.
+type Cache struct {
+	Service string `name:"type" description:"Service used for caching (redis)"`
```
Comment (htdvisser) [inline pkg/config/shared.go]: `name` should probably be `service`

File: cmd/ttn-lw-stack/commands/start.go [hunk 2]
```diff
@@ -45,231 +45,229 @@ import (
...
-
-			if start.IdentityServer || startDefault {
-				logger.Info("Setting up Identity Server")
-				is, err := identityserver.New(c, &config.IS)
-				if err != nil {
-					return shared.ErrInitializeIdentityServer.WithCause(err)
-				}
+			if config.Cache.Service == "redis" {
 				is.SetRedisCache(redis.New(&redis.Config{
 					Redis:     config.Redis,
```
Comment (htdvisser) [inline cmd/ttn-lw-stack/commands/start.go]: Shouldn't this then be `config.Cache.Redis` instead of `config.Redis`?

Comment (KrishnaIyer) [reply cmd/ttn-lw-stack/commands/start.go]: Indeed. Stupid me copied the wrong code while rebasing.

File: pkg/config/shared.go [hunk 3]
```diff
@@ -400,6 +406,7 @@ func (c InteropClient) Fetcher(ctx context.Context, blobConf BlobConfig) (fetch.
 type ServiceBase struct {
 	Base             `name:",squash"`
 	Cluster          Cluster                `name:"cluster"`
+	Cache            Cache                  `name:"Cache"`
```
Comment (htdvisser) [inline pkg/config/shared.go]: @KrishnaIyer I overlooked this during review. The `name` should be lowercase.

Comment (KrishnaIyer) [reply pkg/config/shared.go]: Indeed... :(

**Pull request merged**